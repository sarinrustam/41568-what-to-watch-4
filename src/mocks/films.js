export default [
  {
    id: 0,
    title: `Pulp Fuction`,
    img: `img/pulp-fiction.jpg`
  },
  {
    id: 1,
    title: `Orlando`,
    img: `img/orlando.jpg`
  },
  {
    id: 2,
    title: `Avatar`,
    img: `img/avatar.jpg`
  },
  {
    id: 3,
    title: `No country for old men`,
    img: `no-country-for-old-men.jpg`
  },
  {
    id: 4,
    title: `Aviator`,
    img: `img/aviator.jpg`
  },
  {
    id: 5,
    title: `Macbeth`,
    img: `img/macbeth.jpg`,
  },
  {
    id: 6,
    title: `Johnny English`,
    img: `img/johnny-english.jpg`
  },
  {
    id: 7,
    title: `Seven years in Tibet`,
    img: `img/seven-years-in-tibet`
  },
];
