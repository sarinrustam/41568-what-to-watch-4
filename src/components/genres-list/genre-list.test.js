import React from "react";
import renderer from "react-test-renderer";
import {GENRES} from "../../mocks/films.js";

import GenreList from "./genres-list.jsx";

describe(`Render correctly`, () => {
  it(`should GenreList render correctly`, () => {
    const handlerSetCurrentGenre = jest.fn();
    const tree = renderer.create(
        <GenreList
          genresList={GENRES}
          currentGenre={GENRES[0]}
          setCurrentGenre={handlerSetCurrentGenre}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
