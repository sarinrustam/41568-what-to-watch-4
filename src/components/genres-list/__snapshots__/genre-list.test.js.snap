// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly should GenreList render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Sience
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Action
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Fantasy
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Thriller
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Romance
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Sci-Fi
    </a>
  </li>
</ul>
`;
